prowler/lib/utils/utils.py

```diff
@@ -81,29 +80,26 @@ def hash_sha512(string: str) -> str:
     return sha512(string.encode(encoding_format_utf_8)).hexdigest()[0:9]
 
 
-def is_known_non_secret(line):
-    known_non_secrets = [
-        "MYSQL_ALLOW_EMPTY_PASSWORD",
-        "GCP_SECRET_ARN",
-        "S3_BUCKET",
-        "AWS_STACK",
-    ]
-    for keyword in known_non_secrets:
-        if keyword in line:
-            return True
-    return False
-
-
-def detect_secrets_scan(data=None, file=None):
-    common.is_known_non_secret = is_known_non_secret
-    if not file:
-        temp_data_file = tempfile.NamedTemporaryFile(delete=False)
-        temp_data_file.write(bytes(data, encoding="raw_unicode_escape"))
-        temp_data_file.close()
-
-    secrets = SecretsCollection()
-    with transient_settings(
-        {
+def detect_secrets_scan(data=None, file=None, excluded_secrets: list[str] = None):
```

Comment:
Add the types for the arguments and the return value please.

Reply:
Done